src/main/java/org/infinispan/tutorial/embedded/WeatherApp.java

```diff
@@ -74,21 +74,26 @@ public void shutdown() throws InterruptedException {
 
    public static void main(String[] args) throws Exception {
       WeatherApp app = new WeatherApp();
+      try {
 
-      if (app.cacheManager.isCoordinator()) {
+         if (app.cacheManager.isCoordinator()) {
 
-         app.fetchWeather();
+            app.fetchWeather();
 
-         app.fetchWeather();
+            app.fetchWeather();
 
-         TimeUnit.SECONDS.sleep(5);
+            TimeUnit.SECONDS.sleep(5);
 
-         app.fetchWeather();
+            app.fetchWeather();
 
-         app.computeCountryAverages();
+            app.computeCountryAverages();
+         }
+      } catch (Exception e) {
+         System.out.println("Problem while execution app coordinator : " + e.getMessage());
```

Comment:
I'd probably avoid catching the exception here, since the message printed could remove details that could be useful for other exceptions that potentially be thrown. If no `OWMAPIKEY` is provided, the message is good as is, but for other exceptions it could hide important details.

Reply:
Sorry, I was so busy lately. Now, I have more time. The idea of catching the execption is to reach the finally block to shutdown the app. What do you suggest: to remove the catch or juste the text message ?

Reply:
The finally block would be ran even if there is an exception. You should be fine just making it a try/finally block.

Reply:
I've just pushed changes